readers/hdfs/src/main/java/com/criteo/hadoop/garmadon/hdfs/offset/HdfsOffsetComputer.java

```diff
@@ -45,51 +50,107 @@ public HdfsOffsetComputer(FileSystem fs, Path basePath, @Nullable String kafkaCl
+                if (matcher.matches()) {
+                    try {
+                        int partitionId = Integer.parseInt(matcher.group("partitionId"));
+                        long index = Long.parseLong(matcher.group("index"));
+                        if (isPartitionComputedByThisReader(partitionId, partitionIdsSet) &&
+                            chekCurrentFileIndexBiggerTheOneInHashPartitionDay(partitionId, listedDay, index, resultFile)) {
+                            HashMap<String, FinalEventPartitionFile> dateFinalEventPartitionFile =
+                                (HashMap<String, FinalEventPartitionFile>) resultFile.computeIfAbsent(partitionId, HashMap::new);
+                            dateFinalEventPartitionFile.put(listedDay,
+                                new FinalEventPartitionFile(index, status.getPath()));
+                            resultFile.put(partitionId, dateFinalEventPartitionFile);
+                        }
+                    } catch (NumberFormatException e) {
+                        LOGGER.info("Couldn't deviate a valid offset from '{}'", fileName);
                     }
-                } catch (NumberFormatException e) {
-                    LOGGER.info("Couldn't deviate a valid offset from '{}'", fileName);
                 }
             }
         }
 
+        // Get last offset
+        for (Map.Entry<Integer, Map<String, FinalEventPartitionFile>> partitionIdDateFile : resultFile.entrySet()) {
+            result.put(partitionIdDateFile.getKey(), getMaxOffset(partitionIdDateFile.getValue()));
+        }
+
         return result;
     }
 
-    private String computeDirNamesPattern(LocalDateTime today, int backlogDays) {
-        List<String> dirnames = new ArrayList<>(backlogDays);
+    private boolean isPartitionComputedByThisReader(int partitionId, Set<Integer> partitionIdsSet) {
+        return partitionIdsSet.contains(partitionId);
+    }
+
+    private boolean chekCurrentFileIndexBiggerTheOneInHashPartitionDay(int partitionId, String listedDay, long index, Map<Integer,
+        Map<String, FinalEventPartitionFile>> resultFile) {
+        return isPartitionDayNotAlreadyInHashPartitionDay(partitionId, listedDay, resultFile) || index > resultFile.get(partitionId).get(listedDay).getIndex();
+    }
 
-        for (int i = 0; i < backlogDays; ++i) {
-            LocalDateTime day = today.minusDays(i);
+    private boolean isPartitionDayNotAlreadyInHashPartitionDay(int partitionId, String listedDay, Map<Integer,
+        Map<String, FinalEventPartitionFile>> resultFile) {
+        return !(resultFile.containsKey(partitionId) && resultFile.get(partitionId).containsKey(listedDay));
+    }
 
-            dirnames.add(computeDirName(day));
-        }
+    protected Long getMaxOffset(Map<String, FinalEventPartitionFile> dateFinalEventPartitionFile) {
+        List<Long> maxOffsets = new ArrayList<>();
+        dateFinalEventPartitionFile.values().stream()
```

Comment:
regarding my last comment around stream and parallelism. It is usually more suited to big list processing with ForkJoin pool but we could give it a try. Just few things that needs to be done to allow this:
- get parallelStream instead of stream
- do not put side effect in the lambdas processing the stream. Here it is the fact that we add maxValue to maxOffsets from within the lambda. If we go parallel, there will be synchronisation issues. The proper way to handle this is to use map then collect instead of foreach to create the maxOffsets list.